### PR TITLE
More assoc array docs.

### PR DESCRIPTION
**Pull request by dereuromark**

### CODE REVIEW

File: src/Database/Query.php [hunk 1]
```diff
@@ -835,12 +835,12 @@ public function rightJoin($table, $conditions = [], $types = [])
      * This is a shorthand method for building joins via `join()`.
      *
      * The arguments of this method are identical to the `leftJoin()` shorthand, please refer
-     * to that methods description for further details.
+     * to that methods' description for further details.
```
Comment (othercorey) [inline src/Database/Query.php]:
```suggestion
     * to that method's description for further details.
```

File: src/Http/Response.php [hunk 2]
```diff
@@ -557,6 +557,7 @@ protected function _setHeader(string $header, string $value): void
     /**
      * Clear header
      *
+     * @phpstan-param non-empty-string $header
      * @param string $header Header key.
      * @return void
```
Comment (othercorey) [inline src/Http/Response.php:562]:
```suggestion
     * @param string $header Header key.
     * @return void
     * @phpstan-param non-empty-string $header
```
We seem to keep these on the bottom. Should we continue?

File: src/Database/Query.php (hunk 1 above)
Comment (dereuromark) [reply src/Database/Query.php]: I think it refers Singular

Comment (othercorey) [reply src/Database/Query.php]: Yea. singular possessive is ` before s.

File: src/Http/Response.php (hunk 2 above)
Comment (dereuromark) [reply src/Http/Response.php:562]: I guess a future sniffer might want to fix that instead of humans :)

Comment (othercorey) [reply src/Http/Response.php:562]: I don't really mind, but keeping @param and @return together is probably a good requirement.

File: src/Database/Query.php (hunk 1 above)
Comment (dereuromark) [reply src/Database/Query.php]: Wut? Did I unlearn English?

**Comment by othercorey** [comment]
Looks good to me.

**Pull request merged**